sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -105,7 +105,13 @@ case class CreateViewCommand(
     }
     val sessionState = sparkSession.sessionState
 
-    if (isTemporary) {
+    // 1) CREATE VIEW: create a temp view when users explicitly specify the keyword TEMPORARY;
+    //                 otherwise, create a permanent view no matter whether the temporary view
+    //                 with the same name exists or not.
+    // 2) ALTER VIEW: alter the temporary view if the temp view exists; otherwise, try to alter
```

Comment:
uh, I got it. : ) Copied what @liancheng wrote one month ago. 

allowExisting|replace|SaveMode
-------------|-------|--------
true|	false	|Ignore
false	|false|ErrorIfExists
false|true|Overwrite
true|true|Overwrite

Let me try it. Thanks!